script.js

```diff
@@ -0,0 +1,338 @@
+        this.activeKeyboard = enLowerCase;
+    }
+  }
+
+  changeLang() {
+    this.keyboard.innerHTML = '';
+    switch (this.activeKeyboard) {
+      case enLowerCase:
+        this.createKeyboard(ruLowerCase);
+        this.activeKeyboard = ruLowerCase;
+        localStorage.setItem('myKeyboardLang', '"ru"');
+        break;
+
+      case enUpperCase:
+        this.createKeyboard(ruUpperCase);
+        this.activeKeyboard = ruUpperCase;
+        localStorage.setItem('myKeyboardLang', '"ru"');
+        break;
+
+      case ruLowerCase:
+        this.createKeyboard(enLowerCase);
+        this.activeKeyboard = enLowerCase;
+        localStorage.setItem('myKeyboardLang', '"en"');
+        break;
+
+      case ruUpperCase:
+        this.createKeyboard(enUpperCase);
+        this.activeKeyboard = enUpperCase;
+        localStorage.setItem('myKeyboardLang', '"en"');
+        break;
+
+      default:
+        this.createKeyboard(enLowerCase);
+        this.activeKeyboard = enLowerCase;
+        localStorage.setItem('myKeyboardLang', 'en');
+    }
+  }
+
+  insertKeyboard(parent) {
+    this.keyboard.classList.add('keyboard');
+
+    const textarea = document.createElement('textarea');
+    textarea.setAttribute('autofocus', '');
+
+    parent.append(textarea);
+    parent.append(this.keyboard);
+
+    this.activeKeyboard = this.lang === 'en' ? enLowerCase : ruLowerCase;
+    this.createKeyboard(this.activeKeyboard);
+
+    document.addEventListener('keydown', (e) => {
+      e.preventDefault();
+      document.querySelectorAll('.key').forEach((item) => {
+        if (e.location === parseInt(item.dataset.location, 10) && (
+          e.keyCode === parseInt(item.dataset.code, 10))) {
+          KeyBoard.highlightKey(item);
+          const start = textarea.selectionStart;
+          const end = textarea.selectionEnd;
+          switch (e.keyCode) {
+            case 13:
+              textarea.setRangeText('\n', textarea.selectionStart, textarea.selectionEnd, 'end');
+              break;
+
+            case 8:
+              textarea.value = textarea.value.slice(0, start - 1)
+              + textarea.value.slice(end, textarea.value.length);
+              textarea.selectionStart = start - 1;
+              textarea.selectionEnd = end - 1;
+              break;
+
+            case 20:
+              this.changeCase();
+              KeyBoard.highlightKey(document.querySelector('[data-code = "20"]'));
+              break;
+
+            case 17:
+              this.ctrlIsPressed = true;
+              if (this.altIsPressed) {
+                this.keyboard.innerHTML = '';
+                this.changeLang();
+                KeyBoard.highlightKey(document.querySelectorAll('[data-code = "17"]')[item.dataset.location - 1]);
+                KeyBoard.highlightKey(document.querySelectorAll('[data-code = "18"]')[item.dataset.location - 1]);
+              }
+              break;
+
+            case 16:
+              this.shiftIsPressed = true;
```

Comment:
You have to prevent this action when `this.shiftIsPressed === true` otherwise you redraw the page every second with the button held down